test/metabase/api/card_test.clj

```diff
@@ -3658,8 +3723,27 @@
                 (is (not (mi/can-read? parent-card)))
                 (is (mi/can-read? allowed-collection))
                 (is (mi/can-read? child-card)))
-              (is (= [[1] [2]]
-                     (mt/rows (process-query-for-card child-card)))))))))))
+              (is (= :blocked
```

Comment:
This test was a little confusing for me because the top-level `testing` block says "Should be able to run a Card with another Card as its source query with just perms for the former" but then we end up testing a lot of different scenarios (perms for neither, then different perms options for one).

Reply:
Thanks for pointing that out, fixed